fdbclient/ServerKnobs.cpp

```diff
@@ -36,6 +36,9 @@ void ServerKnobs::initialize(Randomize randomize, ClientKnobs* clientKnobs, IsSi
 	init( MAX_WRITE_TRANSACTION_LIFE_VERSIONS,     5 * VERSIONS_PER_SECOND ); if (randomize && BUGGIFY) MAX_WRITE_TRANSACTION_LIFE_VERSIONS=std::max<int>(1, 1 * VERSIONS_PER_SECOND);
 	init( MAX_COMMIT_BATCH_INTERVAL,                             2.0 ); if( randomize && BUGGIFY ) MAX_COMMIT_BATCH_INTERVAL = 0.5; // Each commit proxy generates a CommitTransactionBatchRequest at least this often, so that versions always advance smoothly
 	MAX_COMMIT_BATCH_INTERVAL = std::min(MAX_COMMIT_BATCH_INTERVAL, MAX_READ_TRANSACTION_LIFE_VERSIONS/double(2*VERSIONS_PER_SECOND)); // Ensure that the proxy commits 2 times every MAX_READ_TRANSACTION_LIFE_VERSIONS, otherwise the master will not give out versions fast enough
+	init( DEFAULT_VERSION_EPOCH,                          1262307600 ); // 2010-01-01T01:00:00+00:00, default version reference epoch (seconds). This value itself is based off the Unix epoch, and is only used when initializing a new cluster. Existing clusters can modify the version epoch through the special key space.
```

Comment:
It looks like our `timer` call is converting windows time into unix epoch seconds, so we're probably ok.

Reply:
This was another bit of confusion. There appear to be different timer calls. `timer()` is different than `g_network->timer()`, and there also exist monotonic versions of each function. It seems like `timer()` is not the right function as it doesn't seem to be deterministic, but `g_network->timer()` leads to https://github.com/apple/foundationdb/blob/82bbfa8aeeea0800b7f54ef6a3474555350a4250/fdbmonitor/fdbmonitor.cpp#L240-L252 (when I use jump to definition, although maybe this is wrong because this function is in `fdbmonitor.cpp`), and this `timer()` implementation is missing Windows functionality.

Reply:
There are other places using `g_network->timer()` though and Windows is currently working, so I think I must be looking at the incorrect implementation.

Reply:
I think this is the implementation of timer used by `Net2`:

https://github.com/apple/foundationdb/blob/82bbfa8aeeea0800b7f54ef6a3474555350a4250/flow/Platform.actor.cpp#L1866